classes/class-wc-connect-shipping-label.php

```diff
@@ -226,10 +225,48 @@ public function get_all_items( WC_Order $order ) {
 				}
 
 				$refunded_qty = $order->get_qty_refunded_for_item( $item->get_id() );
+				$remaining_quantity = $item['qty'] - absint( $refunded_qty );
+				/**
+				 * The threshold at which we will start batching items together.
+				 * As an example, if a single order item has a quantity 60, which is more than the default threshold
+				 * value of 20 we will start batching together.
+				 */
+				$threshold = apply_filters( 'wc_connect_shipment_item_quantity_threshold', 20 );
+				/**
+				 * Max number of shipments allowed to be created for this item should quantity of this order items
+				 * exceeds `wc_connect_shipment_item_quantity_threshold`
+				 */
+				$max_shipments = apply_filters( 'wc_connect_max_shipments_if_quantity_exceeds_threshold', 5 );
+				/**
+				 * Default quantity per item if the `wc_connect_shipments_item_threshold_quantity` is not exceeded.
+				 */
+				$quantity_per_shipment = 1;
+
+				$weight_per_item = $item_data['weight'];
+				$should_cap_shipments = $remaining_quantity > $threshold;
+
+				if ( $should_cap_shipments ) {
+					$quantity_per_shipment = floor( $remaining_quantity / $max_shipments );
+					for ( $i = 0; $i < $max_shipments; $i ++ ) {
+						$item_data['weight'] = $quantity_per_shipment * $weight_per_item;
```

Comment:
Should we maybe round the weight to 2 decimal places?

![screenshot-localhost_8050-2024 05 07-07_49_56](https://github.com/Automattic/woocommerce-services/assets/11618203/792e56d7-b64c-4031-b91d-f138f2017eaf)